### ISSUE TITLE
Application is too slow in general

### ISSUE THREAD

**Issue by mbedded** [Microsoft/azuredatastudio#2849]
Issue Type: <b>Bug</b>

- Missing Intellisense for tables/columns
- "select top 1000 rows" needs 10 seconds to create a tab and execute the quers
- Overall very slow reaction times. Many actions need a lot of more time than in MSSQL-Management Studio

Azure Data Studio version: azuredatastudio 1.0.0 (cab8f3e37c9ac4367e0dacef5185da08bd8021c2, 2018-09-20T22:04:35.897Z)
OS version: Windows_NT x64 10.0.16299

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz (4 x 2304)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|7.87GB (1.08GB free)|
|Process Argv|C:\Program Files\Azure Data Studio\azuredatastudio.exe|
|Screen Reader|no|
|VM|0%|

</details>Extensions: none

**Comment by Poltergeisen** [comment]
I'm having an issue with a query that is 5mb in size. It was a generated data extract script from someone with SSMS. It completely lagged out my window and the script wouldn't run.

**Comment by aaronsmith1234** [comment]
I'm also having issues, simple queries can take a long time to run

**Comment by git-xbits** [comment]
I'm also having problems with connection (pre-login), error with connection in new Query tab and after error with slow connection.

In MSSMS 18 or before runs perfectly.

**Comment by mbedded** [comment]
For the "new-posters": Could you add your current version pls and system information? My ticket is from Oct 2018 with "azuredatastudio 1.0.0".

My issue was resolved by a restart and didn't occurred once since them. I guess the software or my pc had just a bad day. I just left it open because of the "intellisense" part. Maybe (because we have two different issues here) we should split it into two separate tickets.

Greets

**Comment by git-xbits** [comment]
Same here, 2 day to now I have not any issue. I don't know why ahhaha

**Comment by pnhughes** [comment]
Version details appear at bottom of post.

I experience **typing lag** from 1 - 3 seconds in both text and code blocks when using notebooks in the application. Lag starts to appear after notebooks have grown sufficiently but I cannot identify a size threshold at present.  Current notebook is 6 text blocks and 7 code blocks, with all output cleared.  14,260 bytes total for the file, 1,942 words, 89 lines. Typing lag has just started.  

Query performance does not appear to be affected, and I have had no performance issues, even with larger queries.  

Happy to perform additional testing if anyone has any direction/suggestions.

ADS info:
Version: 1.9.0 (user setup)
Commit: 78a42e1d112ae3231777722b51eaf44f83ddbe55
Date: 2019-07-10T04:31:36.998Z
VS Code 1.34.0
Electron: 3.1.8
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32

System Info:
Surface Pro 6
Intel Core i5-8350 CPU @ 1.70 GHz
8 GB RAM
OS: Windows_NT x64 10.0.17763

**Comment by nioertel** [comment]
Similar issue here.
Connecting to a server takes around 60 seconds (simple SQL Server auth).
Executing a query takes even longer. Query window appears after 30 seconds (SELECT TOP 1000), execution and loading of the results takes at least another 30 seconds.
I tested with simple table (15 columns; 18 rows; data types: int, bigint, varchar, nvarchar, datetime).

```
Version: 1.15.1
Commit: 83544eba2504f564d785534cc780d2073feec2cc
Date: 2020-02-15T05:01:39.777Z
VS Code: 1.42.0
Electron: 6.1.6
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Darwin x64 18.7.0
```
I'm using a Macbook Pro (15" 2018; 2,2 GHz Intel Core i7; 32 GB 2400 MHz DDR4).

**Comment by kasbert** [comment]
Disabling Settings / Application / Telemetry made a big difference for me.

**Comment by ponchautf** [comment]
> Disabling Settings / Application / Telemetry made a big difference for me.

Yes, Data studio is faster with telemetry disabled but queries are still slower than with other application 
e.g. with SQLPRO studio connection to DB is less than 1sec (5-15 second with ADS) and execute queries are faster too, getting 1000 rows is about 1.214 sec , and 5-20 seconds in ADS (can't be more precise, the time for the query is always 00:00:00 !!)
